packages/shared/src/lib/placeholders.ts

```diff
@@ -1,20 +1,90 @@
-import { getCountryName, getLocalCurrencySymbol } from './geolocation';
+import { getCountryName, getLocalCurrencySymbol, countryCodeToCountryGroupId } from './geolocation';
+import { Prices } from '../types/prices';
+
+// TODO: determine if we want to carry default pricing in the code
+// - if yes, then there may be a more obvious place to define this data structure
+// - if no, then may be useful to keep this here as it will tell us that something has gone wrong with the prices retrieval code or cache
```

Comment:
@tomrf1 - what are your feelings about hardcoded default prices? My concern is that if the pricing data fetch fails, or the caching fails, we risk showing incorrect price data to users. But without default pricing we risk showing the user a meaningless message.

Reply:
I do not think we want to maintain hardcoded default prices. It's better to show nothing and trigger an alarm if prices are unavailable.

I think there are 2 approaches:
1. fail client-side. If the `prices` prop is undefined and the copy has a price template then render nothing ([precedent here](https://github.com/guardian/support-dotcom-components/blob/main/packages/modules/src/modules/banners/common/BannerWrapper.tsx#L181)). But we'd need to trigger a new alarm server-side as well.
2. fail server-side. If the prices are unavailable, return a 500 status for banner requests until they're available again. This would mean no banners are displayed, but at least we'd get an alarm very quickly (from all the 500s).

I think 1 is preferable as it allows e.g. the contributions banner to continue working when prices are unavailable.

Reply:
Done - but I'm not sure how the `reject` results tie in with caching. Needs close review.

Reply:
memoizee will not cache failed promises.
Given that the `prices` prop is optional and this eventuality is handled client-side, I suggest adding a `catch` that just sets it to `undefined`